test/SourceKit/Indexing/index_effective_access_level.swift.response

```diff
@@ -0,0 +1,260 @@
+      key.entities: [
+        {
+          key.kind: source.lang.swift.ref.enum,
+          key.name: "InternalEn",
+          key.usr: "s:28index_effective_access_level10InternalEnO",
+          key.line: 27,
+          key.column: 11,
+          key.effective_access_level: source.decl.attribute.internal
+        },
+        {
+          key.kind: source.lang.swift.decl.function.method.instance,
+          key.name: "foo()",
+          key.usr: "s:28index_effective_access_level10InternalEnO3fooyyF",
+          key.line: 28,
+          key.column: 10,
+          key.effective_access_level: source.decl.attribute.internal
+        }
+      ],
+      key.effective_access_level: source.decl.attribute.internal
+    },
+    {
+      key.kind: source.lang.swift.decl.extension.enum,
+      key.name: "PrivateEn",
+      key.usr: "s:e:s:28index_effective_access_level9PrivateEn33_2295DDF1454D6A6D9229E8222CD85214LLO5prFooyyF",
+      key.line: 31,
+      key.column: 19,
+      key.entities: [
+        {
+          key.kind: source.lang.swift.ref.enum,
+          key.name: "PrivateEn",
+          key.usr: "s:28index_effective_access_level9PrivateEn33_2295DDF1454D6A6D9229E8222CD85214LLO",
+          key.line: 31,
+          key.column: 19,
+          key.effective_access_level: source.decl.attribute.fileprivate
+        },
+        {
+          key.kind: source.lang.swift.decl.function.method.instance,
+          key.name: "prFoo()",
+          key.usr: "s:28index_effective_access_level9PrivateEn33_2295DDF1454D6A6D9229E8222CD85214LLO5prFooyyF",
+          key.line: 32,
+          key.column: 10,
+          key.effective_access_level: source.decl.attribute.fileprivate
+        }
+      ],
+      key.attributes: [
+        {
+          key.attribute: source.decl.attribute.private
+        }
+      ],
+      key.effective_access_level: source.decl.attribute.fileprivate
+    },
+    {
+      key.kind: source.lang.swift.decl.extension.enum,
+      key.name: "PrivateEn",
+      key.usr: "s:e:s:28index_effective_access_level9PrivateEn33_2295DDF1454D6A6D9229E8222CD85214LLO5prFooyXeXeF",
+      key.line: 35,
+      key.column: 11,
+      key.entities: [
+        {
+          key.kind: source.lang.swift.ref.enum,
+          key.name: "PrivateEn",
+          key.usr: "s:28index_effective_access_level9PrivateEn33_2295DDF1454D6A6D9229E8222CD85214LLO",
+          key.line: 35,
+          key.column: 11,
+          key.effective_access_level: source.decl.attribute.fileprivate
+        },
+        {
+          key.kind: source.lang.swift.decl.function.method.instance,
+          key.name: "prFoo()",
+          key.usr: "s:28index_effective_access_level9PrivateEn33_2295DDF1454D6A6D9229E8222CD85214LLO5prFooyXeXeF",
+          key.line: 36,
+          key.column: 18,
+          key.attributes: [
+            {
+              key.attribute: source.decl.attribute.private
+            }
+          ],
+          key.effective_access_level: source.decl.attribute.fileprivate
```

Comment:
This is incorrect: the effective visibility of `extension PrivateEn { private func prFoo() {} }` is less than `fileprivate`. You can prove this for yourself by trying to call that method (you would need to give your duplicated methods different names first) from a free function in the same file:

```swift
private enum E {
    case a, b, c
}
extension E {
    private func f() { print("Hello") }
}
E.a.f() // error: 'f' is inaccessible due to 'private' protection level
```

And this is what I've been saying earlier: there is no unambiguous way to distinguish this `private` scope from a totally different `private` scope. Without more radically designing your output, all you can get here is whether a declaration is effectively `public`, `internal`, `fileprivate`, or _something less than that_. I would not reuse `source.decl.attribute.private` to describe everything in the latter category, as it's misleading (it would make every `private` scope appear the same).

Reply:
I am confused on how to proceed because that output came directly from `getEffectiveAccess()`. Do you mean that ideally the compiler should have an additional enum case for the latter category?

Reply:
Right, so `getEffectiveAccess()` isn't giving you the right information here. That is also what it says in the documentation: "This is the access used when making optimization and code generation decisions. It should not be used at the AST or semantic level."

Reply:
I think I understood that we should never use `attribute.private` to describe `AccessLevel.private`, but It's unclear to me the strategy to make the compiler stop returning `fileprivate` in this case (should it have returned `private`?)

Reply:
> Right, so `getEffectiveAccess()` isn't giving you the right information here. That is also what it says in the documentation: "This is the access used when making optimization and code generation decisions. It should not be used at the AST or semantic level."

Oh sorry, I missed that. Using this was probably a bad suggestion on my part... Basing this on `getFormalAccessScope().isPublic/Private/FileScope/Internal()` instead is probably a better fit and I think will give the correct result here. Just remember to check `isFileScope()` (to indicate `fileprivate`) before `isPrivate()`, because `private` things accessible at file scope are effectively `fileprivate`.